src/app/OperationalSessionSetup.cpp

```diff
@@ -201,6 +213,20 @@ void OperationalSessionSetup::UpdateDeviceData(const ResolveResult & result)
 {
     auto & config = result.mrpRemoteConfig;
     auto addr     = result.address;
+    UpdateDeviceData(addr, config, result.supportsTcpServer);
+}
+
+void OperationalSessionSetup::UpdateDeviceData(const Transport::PeerAddress & addr, const ReliableMessageProtocolConfig & config, bool supportsTcpServer)
+{
+    // If we are in the process of trying to establish a session, we need to
+    // cancel that attempt before we can update the address.
+    if (mState == State::Connecting)
+    {
+        CancelSessionSetupReattempt();
```

Comment:
CancelSessionSetupReattempt() does not in fact take us out of the Connecting state, right?  It just cancels a timer...